pkg/kubelet/pod/pod_manager.go

```diff
@@ -77,10 +78,16 @@ type Manager interface {
 	// this means deleting the mappings related to mirror pods.  For non-
 	// mirror pods, this means deleting from indexes for all non-mirror pods.
 	DeletePod(pod *v1.Pod)
-	// DeleteOrphanedMirrorPods deletes all mirror pods which do not have
-	// associated static pods. This method sends deletion requests to the API
-	// server, but does NOT modify the internal pod storage in basicManager.
-	DeleteOrphanedMirrorPods()
+	// GetOrphanedMirrorPodNames returns names of orphaned mirror pods
```

Comment:
The concept of orphaned mirror pod refers to mirror pod which is in api server but which is no longer backed by file in static pod manifest source directory.

For GetOrphanedMirrorPodNames(), it seems we don't need to mention graceful termination since it is the responsibility of orphan pod cleaner (located in kubelet_pods.go) to clean up mirror pods (respecting grace period specified for the underlying static pod).

w.r.t. the suggestion of removing relevant podByFullName entry in MarkMirrorPodTerminated, I have looked at the func's which depend on the value for podByFullName entry.

Let's look at the following call in Kubelet#GetAttach:
```
	pod, found := kl.GetPodByFullName(podFullName)
	if !found || (string(podUID) != "" && pod.UID != podUID) {
```
Currently, when the static pod is deleted, GetPodByFullName() would return false for ```found```, returning early.
If we remove podByFullName entry in MarkMirrorPodTerminated, the above call would return true for ```found```,  likely resulting in proceeding to remaining part of Kubelet#GetAttach().

I think we should remove podByFullName entry in DeletePod() because there may be existing code path(s) which depend on pod not being returned from GetPodByFullName() when the pod is undergoing termination.

Please let me know if you agree.

Reply:
i agree that if we evolve pod manager to track pods undergoing termination, then we need to keep the behavior you have.  i think it makes it harder to understand what pod manager is itself doing, and so that is why i was wondering if we could get to a simpler resolution that checks if (mirror,pod) is still present in pod killing channel.